lib/matplotlib/patches.py

```diff
@@ -3218,23 +3218,88 @@ class _Curve(_Base):
+        or closed.
         """
 
         beginarrow = endarrow = False  # Whether arrows are drawn.
         fillbegin = fillend = False  # Whether arrows are filled.
 
-        def __init__(self, head_length=.4, head_width=.2):
+        def __init__(self, head_length=.4, head_width=.2, widthA=1., widthB=1.,
+                     lengthA=0.2, lengthB=0.2, angleA=0, angleB=0, scaleA=None,
+                     scaleB=None):
             """
             Parameters
             ----------
             head_length : float, default: 0.4
                 Length of the arrow head, relative to *mutation_scale*.
             head_width : float, default: 0.2
                 Width of the arrow head, relative to *mutation_scale*.
+            widthA : float, default: 1.0
+                Width of the bracket at the beginning of the arrow
+            widthB : float, default: 1.0
+                Width of the bracket at the end of the arrow
+            lengthA : float, default: 0.2
+                Length of the bracket at the beginning of the arrow
+            lengthB : float, default: 0.2
+                Length of the bracket at the end of the arrow
+            angleA : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            angleB : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            scaleA : float, default *mutation_size*
+                The mutation_size for the beginning bracket
+            scaleB : float, default *mutation_size*
+                The mutation_size for the end bracket
             """
+
+            self.beginarrow_head = False
+            self.beginarrow_bracket = False
+            self.endarrow_head = False
+            self.endarrow_bracket = False
             self.head_length, self.head_width = head_length, head_width
+            self.widthA, self.widthB = widthA, widthB
+            self.lengthA, self.lengthB = lengthA, lengthB
+            self.angleA, self.angleB = angleA, angleB
+            self.scaleA, self.scaleB = scaleA, scaleB
+
+            if self.beginarrow == 'arrow':
+                self.beginarrow_head = True
+                self.beginarrow_bracket = False
+            elif self.beginarrow == 'bracket':
+                self.beginarrow_head = False
+                self.beginarrow_bracket = True
+            elif self.beginarrow:
+                self.beginarrow_head = True
+                self.beginarrow_bracket = False
+
+                import logging
+                _log = logging.getLogger(__name__)
+                _log.info("The use of boolean values in beginarrow is now "
+                          "decapitated, prefer using the string ‘arrow’ "
+                          "for the same function")
+
```

Comment:
You should check explicitly if it is of type `bool` and raise a warning using `_api.warn_deprecated` for both `True` and `False`. And if it's _not_ a `bool`, the last `else` can use `_api.check_in_list` to raise an error for everything else. The default should probably be `'none'` instead of `False`, and the new attributes should be made private.